extensions/cornerstone-dynamic-volume/src/panels/PanelGenerateImage.tsx

```diff
@@ -0,0 +1,414 @@
+                const options = numTimePointsToOptions(
+                  volumeData.volume._numTimePoints
+                );
+                const range = [1, volumeData.volume._numTimePoints];
+                setTimeOptions(prevArray => [...prevArray, ...options]);
+                setRangeValues(prevArray => [...prevArray, ...range]);
+                setSliderValues(range);
+                setTimeFramesToUse([0, 1]);
+                const computedVolumeInit = createComputedVolume(
+                  volumeData.volume.volumeId,
+                  computedVolumeId
+                );
+                setMyComputedVolume(computedVolumeInit);
+              }
+            }
+          });
+        }).unsubscribe
+      );
+    });
+
+    return () => {
+      subscriptions.forEach(unsub => {
+        unsub();
+      });
+    };
+  }, []);
+
+  // Get computed volume from cache, calculate the data across the time frames,
+  // set the scalar data to the computedVolume, and create displaySet
+  function onGenerateImage() {
+    const computedVolume = cache.getVolume(computedVolumeId);
+    const dataInTime = cstUtils.dynamicVolume.generateImageFromTimeData(
+      myDynamicVolume,
+      generateOptions.Operation,
+      timeFramesToUse
+    );
+
+    // Add loadStatus.loaded to computed volume and set to true
+    computedVolume.loadStatus = {};
+    computedVolume.loadStatus.loaded = true;
+
+    // Set computed scalar data to volume
+    const scalarData = computedVolume.getScalarData();
+    for (let i = 0; i < dataInTime.length; i++) {
+      scalarData[i] = dataInTime[i];
+    }
+
+    // If computed display set does not exist, create an object to be used as
+    // the displaySet. If it does exist, update the image data and vtkTexture
+    if (!computedDisplaySet) {
+      const obj = {
+        [uuidComputedVolume]: {
+          volumeLoaderSchema: computedVolume.volumeId.split(':')[0],
+          displaySetInstanceUID: uuidComputedVolume,
+          SOPClassHandlerId:
+            '@ohif/extension-default.sopClassHandlerModule.stack',
+          SOPClassUID: '1.2.840.10008.5.1.4.1.1.128',
+          SeriesInstanceUID: 'test',
+          StudyInstanceUID: 'test',
+          SeriesNumber: 0,
+          FrameRate: undefined,
+          SeriesDescription: '',
+          Modality: myDynamicVolume.metadata.Modality,
+          isMultiFrame: false,
+          countIcon: undefined,
+          numImageFrames: 1,
+          uid: uuidComputedVolume,
+          averageSpacingBetweenFrames: null,
+        },
+      };
+      setComputedDisplaySet(obj);
+    } else {
+      // update vtkOpenGLTexture and imageData of computed volume
+      const { imageData, vtkOpenGLTexture } = computedVolume;
+      const numSlices = imageData.getDimensions()[2];
+      const slicesToUpdate = [...Array(numSlices).keys()];
+      slicesToUpdate.forEach(i => {
+        vtkOpenGLTexture.setUpdatedFrame(i);
+      });
+      imageData.modified();
```

Comment:
This is a usefull method, put it in commands module of cornerstone extension and call it updateVolumeTextureAndImageData or similar and run that method with commandsManager here